ProcessLib/LiquidFlow/LiquidFlowLocalAssembler-impl.h

```diff
@@ -122,40 +144,61 @@ void LiquidFlowLocalAssembler<ShapeFunction, IntegrationMethod, GlobalDim>::
     ParameterLib::SpatialPosition pos;
     pos.setElementID(_element.getID());
 
-    // TODO: The following two variables should be calculated inside the
-    //       the integration loop for non-constant porosity and storage models.
-    double porosity_variable = 0.;
-    double storage_variable = 0.;
+    auto const& medium =
+        _process_data.media_map->getMedium(this->_element.getID());
+    auto const& liquid_phase = medium->phase("AqueousLiquid");
+    auto const& solid_phase = medium->phase("Solid");
+
+    MaterialPropertyLib::VariableArray vars;
+
     for (unsigned ip = 0; ip < n_integration_points; ip++)
     {
         auto const& ip_data = _ip_data[ip];
 
         double p = 0.;
         NumLib::shapeFunctionInterpolate(local_x, ip_data.N, p);
+        vars[static_cast<int>(MaterialPropertyLib::Variable::phase_pressure)] =
+            p;
+
+        // Compute density:
+        auto const fluid_density =
+            liquid_phase.property(MaterialPropertyLib::PropertyType::density)
+                .template value<double>(vars, pos, t, dt);
+        auto const ddensity_dpressure =
+            liquid_phase.property(MaterialPropertyLib::PropertyType::density)
+                .template dValue<double>(
+                    vars, MaterialPropertyLib::Variable::phase_pressure, pos,
+                    t, dt);
+
+        assert(fluid_density > 0.);
```

Comment:
It can be moved to the place after the definition of `fluid_density`.

Reply:
I'll move it as suggested.